scheduler.py

```diff
@@ -154,6 +154,12 @@ def _run(dryrun=False):
     The main loop.  Iterate over our report config and try to run reports that
     are scheduled to run now.
     """   
+    try:
+        me = singleton.SingleInstance()
+    except SystemExit:
+        print "** Quitting run at %s" % datetime.now().isoformat()   
+        sys.exit(-1)
+        
     run_logger = RunLogger()
     logging.config.dictConfig(LOGGING)
```

Comment:
@currycoder does in here work better? 
surrounding just the singleton line works as well.